File: LeetCode/problems/2181-merge-nodes-in-between-zeros/2181-merge-nodes-in-between-zeros.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeNodes(self, head: Optional[ListNode]) -> Optional[ListNode]:
        pre = head
        cur = head.next
        score = 0
        while(cur):
            if cur.val == 0:
                pre.val = score
                score = 0
                if cur.next:
                    pre = pre.next
                cur = cur.next
            else:
                score += cur.val
                cur = cur.next

        pre.next = None
        return head

File: LeetCode/problems/2181-merge-nodes-in-between-zeros/2181-merge-nodes-in-between-zeros.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var mergeNodes = function(head) {
    let prev = head;
    let cur = head.next;
    let score = 0;

    while(cur){
        if(cur.val==0){
            prev.val = score;
            score = 0;
            if(cur.next !==null){
                prev.next = cur;
                prev = cur;
            }else{
                prev.next = null;
            }
            cur = cur.next;
        }else{
            score += cur.val;
            cur = cur.next;
        }
    }
    return head;
};